torch/lib/THCUNN/SpatialDepthwiseConvolution.cu

```diff
@@ -85,28 +107,39 @@ __global__ void spatialDepthwiseConvolutionUpdateGradInput(
 {
   const int channelStride = inputHeight * inputWidth;
   const int batchStride = inputChannels * channelStride;
+  const int KW_LIMIT = (kSize !=0) ? kSize : kernelWidth;
+  const int KH_LIMIT = (kSize !=0) ? kSize : kernelHeight;
+  const int strideW = (stride !=0) ? stride : strideWidth;
+  const int strideH = (stride !=0) ? stride : strideHeight;
 
   for (IndexType linearIndex = blockIdx.x * blockDim.x + threadIdx.x;
        linearIndex < totalElements;
        linearIndex += gridDim.x * blockDim.x) {
 
-    const int n = linearIndex / batchStride;
-    const int c = (linearIndex / channelStride) % inputChannels;
-    const int h = (linearIndex / inputWidth) % inputHeight;
-    const int w = linearIndex % inputWidth;
+    int indtmp1 = linearIndex/inputWidth;
+    const int w = linearIndex - indtmp1 * inputWidth;
+    int indtmp2 = indtmp1/inputHeight;
+    const int h = indtmp1 - indtmp2 * inputHeight;
+    indtmp1 = indtmp2; 
+    indtmp2 = indtmp1/inputChannels;
+    const int c = indtmp1 - indtmp2 * inputChannels;
+    const int n = indtmp2;
```

Comment:
Can you include the original code in a comment (just to conceptually know what happens here)? This is really hard to read

Reply:
Will do